test/e2e/shared/openstack.go

```diff
@@ -485,6 +485,42 @@ func GetOpenStackServers(e2eCtx *E2EContext, openStackCluster *infrav1.OpenStack
 	return srvs, nil
 }
 
+// GetOpenStackServer returns the server with the given ID along with the first
+// IP address it has in the OpenStackCluster network.
+func GetOpenStackServerWithIP(e2eCtx *E2EContext, id string, openStackCluster *infrav1.OpenStackCluster) (*ServerExtWithIP, error) {
+	providerClient, clientOpts, _, err := GetTenantProviderClient(e2eCtx)
+	if err != nil {
+		_, _ = fmt.Fprintf(GinkgoWriter, "error creating provider client: %s\n", err)
+		return nil, nil
+	}
+
+	computeClient, err := clients.NewComputeClient(providerClient, clientOpts)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create compute client: %w", err)
+	}
+	srvExt, err := computeClient.GetServer(id)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get server: %w", err)
+	}
+
+	instanceStatus := compute.NewInstanceStatusFromServer(srvExt, logr.Discard())
+	instanceNS, err := instanceStatus.NetworkStatus()
+	if err != nil {
+		return nil, fmt.Errorf("error getting network status for server %s: %v", srvExt.Name, err)
+	}
+
+	ip := instanceNS.IP(openStackCluster.Status.Network.Name)
+	if ip == "" {
+		return nil, fmt.Errorf("error getting internal ip for server %s: internal ip doesn't exist (yet)", srvExt.Name)
+	}
+	srvExtWithIP := ServerExtWithIP{
+		ServerExt: *srvExt,
+		ip:        ip,
+	}
+
+	return &srvExtWithIP, nil
```

Comment:
What is the reason for returning a pointer here?

Reply:
Good question! I think I just mimicked the function above. It would make sense to remove the pointer here. :+1:

Reply:
Fixed!